PowerEditor/src/NppIO.cpp

```diff
@@ -334,7 +334,7 @@ BufferID Notepad_plus::doOpen(const generic_string& fileName, bool isRecursive,
 	BufferID buffer;
 	if (isSnapshotMode)
 	{
-		buffer = MainFileManager.loadFile(longFileName, NULL, encoding, backupFileName, fileNameTimestamp);
+		buffer = MainFileManager.loadFile(longFileName, Document{}, encoding, backupFileName, fileNameTimestamp);

```

Comment:
Could you provide the reference about the empty curly braces notation in c++?

Reply:
Yes. It is called "uniform initialization":
https://isocpp.org/wiki/faq/cpp11-language#uniform-init
When there is nothing inside the curly braces, a proper default value is used. 
It doesn't matter whether it's a built-in integral type (`0`) or some class (`std::string("")`).

In this case, Document is defined as following:
`typedef sptr_t Document;`
But using uniform initialization, we don't care about the actual type.

Do you think it's better to use `0`?

Reply:
OK, thank you for the info.
Please remove all the empty curly braces notation - it's not explicit enough. 
I prefer that pointers are initialized by a `NULL` or `nullptr`, but not `0` or `{}`.

Reply:
`Document` is integer but not a pointer. Changed to 0.

In [Buffer.h](https://github.com/notepad-plus-plus/notepad-plus-plus/blob/master/PowerEditor/src/ScintillaComponent/Buffer.h#L28):
`typedef sptr_t Document;`

In [Scintilla.h](https://github.com/notepad-plus-plus/notepad-plus-plus/blob/master/scintilla/include/Scintilla.h#L35):
`typedef intptr_t sptr_t;`

In `<stdint.h>`:
`typedef __int64          intptr_t;`

Reply:
> Document is integer but not a pointer. Changed to 0.

Yes, I know. I checked it.
But with the same logic, every pointers are an integer, period.